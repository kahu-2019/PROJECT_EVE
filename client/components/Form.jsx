import React from "react";
import { getEmail } from "../api/index";

export default class SubmitForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      breach: []
    };
    this.handleChange = this.handleChange.bind(this);
    this.clickHandle = this.clickHandle.bind(this);
  }

  handleChange(event) {
    this.setState({
      breach: [event.target.value]
    });
  }
  clickHandle() {
    return getEmail(this.state.breach).then(res =>
      this.setState({ breach: res })
    );
  }

  render() {
    return (
      <div className="formContainer">
        <form onSubmit={event => event.preventDefault()}>
          <label>
            <input
              type="text"
              email="text"
              onChange={this.handleChange}
              placeholder="enter your email"
            />
          </label>
          <button className="submitButton" onClick={this.clickHandle}>
            Submit
          </button>
        </form>

        <ul>{this.state.breach.map((obj, i) => <li key={i}>{obj.Domain}: <p>{obj.DataClasses} </p></li>)}</ul>
      </div>
    );
  }
}
