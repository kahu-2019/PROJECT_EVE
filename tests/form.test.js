import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Form from "../client/components/Form";

describe("Form should render", () => {
  it("should render correctly", () => {
    const component = shallow(<Form />);

    expect(component).toMatchSnapshot();
  });
});
