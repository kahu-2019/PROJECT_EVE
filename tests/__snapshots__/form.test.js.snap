// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubmitForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
        >
          <label>
            <input
              email="text"
              onChange={[Function]}
              placeholder="enter your email"
              type="text"
            />
          </label>
          <button
            className="submitButton"
            onClick={[Function]}
          >
            Submit
          </button>
        </form>,
        <ul />,
      ],
      "className": "formContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              <input
                email="text"
                onChange={[Function]}
                placeholder="enter your email"
                type="text"
              />
            </label>,
            <button
              className="submitButton"
              onClick={[Function]}
            >
              Submit
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                email="text"
                onChange={[Function]}
                placeholder="enter your email"
                type="text"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "email": "text",
                "onChange": [Function],
                "placeholder": "enter your email",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "submitButton",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
          >
            <label>
              <input
                email="text"
                onChange={[Function]}
                placeholder="enter your email"
                type="text"
              />
            </label>
            <button
              className="submitButton"
              onClick={[Function]}
            >
              Submit
            </button>
          </form>,
          <ul />,
        ],
        "className": "formContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                <input
                  email="text"
                  onChange={[Function]}
                  placeholder="enter your email"
                  type="text"
                />
              </label>,
              <button
                className="submitButton"
                onClick={[Function]}
              >
                Submit
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  email="text"
                  onChange={[Function]}
                  placeholder="enter your email"
                  type="text"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "email": "text",
                  "onChange": [Function],
                  "placeholder": "enter your email",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "submitButton",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
